### PR TITLE
hv: remove CONFIG_HV_RAM_SIZE

### PR DESCRIPTION
**Pull request by lifeix**
It's difficult to configure CONFIG_HV_RAM_SIZE properly at once. This patch
not only remove CONFIG_HV_RAM_SIZE, but also we use ld linker script to
dynamically get the size of HV RAM size.

Tracked-On: #6663
Signed-off-by: Fei Li <fei1.li@intel.com>
Acked-by: Eddie Dong <eddie.dong@intel.com>

### CODE REVIEW

**Review by dongyaozu — APPROVED** [review]

**Comment by dongyaozu** [comment]
@yakuizhao

**Review by dbkinder — CHANGES_REQUESTED** [review]
I suspect there are additional changes needed.

File: hypervisor/include/arch/x86/asm/boot/ld_sym.h [hunk 1]
```diff
@@ -14,5 +14,6 @@ extern uint8_t		ld_entry_end;
 extern const uint8_t	ld_trampoline_load;
 extern uint8_t		ld_trampoline_start;
 extern uint8_t		ld_trampoline_end;
+extern uint8_t		_ld_ram_size;
```
Comment (junjiemao1) [inline hypervisor/include/arch/x86/asm/boot/ld_sym.h:17]: Our coding guideline does not recommend variables starting with underscores. See https://projectacrn.github.io/latest/developer-guides/c_coding_guidelines.html#c-id-07-names-defined-by-developers-shall-not-start-with-underscore

**Comment by dbkinder** [comment]
@junjiemao1  A third option could be to change the configurator (or if needed, add a new tool) to remove options it doesn't recognize and add options it notices are missing in the imported XML so it can be saved as "fixed", based on the schema definition (xsd) files for the release.

File: hypervisor/include/arch/x86/asm/boot/ld_sym.h (hunk 1 above)
Comment (lifeix) [reply hypervisor/include/arch/x86/asm/boot/ld_sym.h]: done

**Review by yakuizhao — APPROVED** [review]
LGTM

**Comment by wenlingz** [comment]
> > I suspect there are additional code and doc changes needed.
> 
> Hi David, this patch only does the change in HV side. Nanlin would find someone to help do the change in config tool side as well as the doc side.

merge firstly, @NanlinXie please help add doc part in future, thanks

**Pull request merged**